lib/livebook_web/live/session_live/app_teams_component.ex

```diff
@@ -0,0 +1,232 @@
+        which deployment group you want to send this app and then click the button to Deploy.
+      </p>
+
+      <div class="flex gap-12">
+        <p class="text-gray-700">
+          <.label>Hub</.label>
+          <span>
+            <span class="text-lg"><%= @hub.hub_emoji %></span>
+            <span><%= @hub.hub_name %></span>
+          </span>
+        </p>
+        <%= if @deployment_groups do %>
+          <%= if @deployment_groups != [] do %>
+            <.form
+              for={@deployment_group_form}
+              phx-change="select_deployment_group"
+              phx-target={@myself}
+              id="select_deployment_group_form"
+            >
+              <.select_field
+                help={deployment_group_help()}
+                field={@deployment_group_form[:deployment_group_id]}
+                options={deployment_group_options(@deployment_groups)}
+                label="Deployment Group"
+                name="deployment_group_id"
+                value={@deployment_group_id}
+              />
+            </.form>
+          <% else %>
+            <p class="text-gray-700">
+              <.label help={deployment_group_help()}>
+                Deployment Group
+              </.label>
+              <span>No deployment groups available</span>
+            </p>
+          <% end %>
+        <% end %>
+      </div>
+
+      <.message_box :if={@message} kind={@message_kind} message={@message} />
+
+      <div :if={@app_deployment} class="space-y-3 pb-4">
+        <p class="text-gray-700">Current deployed version:</p>
+
+        <ul class="text-gray-700 space-y-3">
+          <li class="flex gap-2">
+            <div class="font-bold">Title:</div>
+            <span><%= @app_deployment.title %></span>
+          </li>
+          <li class="flex gap-2">
+            <div class="font-bold">Deployed by:</div>
+            <span><%= @app_deployment.deployed_by %></span>
+          </li>
+          <li class="flex gap-2">
+            <div class="font-bold">Deployed at:</div>
+            <span><%= @app_deployment.deployed_at %></span>
+          </li>
+        </ul>
+      </div>
+
+      <.button
+        id="deploy-livebook-agent-button"
+        color="blue"
+        phx-click="deploy_app"
+        phx-target={@myself}
+      >
+        <.remix_icon icon="rocket-line" /> Deploy to Livebook Agent
+      </.button>
+    </div>
+    """
+  end
+
+  @impl true
+  def handle_event("select_deployment_group", %{"deployment_group_id" => id}, socket) do
+    id = if(id != "", do: id)
+    Livebook.Session.set_notebook_deployment_group(socket.assigns.session.pid, id)
+
+    {:noreply, socket}
+  end
+
+  def handle_event("deploy_app", _, socket) do
+    session = socket.assigns.session
+    deployment_group = socket.assigns.deployment_group
+    app_settings = socket.assigns.app_settings
+
+    data = Livebook.Session.get_data(session.pid)
```

Comment:
We only need the notebook, so let's use `Livebook.Session.get_notebook` to copy less data :)